modules/update-cvo-conditions.adoc

```diff
@@ -0,0 +1,65 @@
+// Module included in the following assemblies:
+//
+// * updating/understanding-update-service.adoc
+
+[id="update-cvo-conditions_{context}"]
+= Understanding CVO conditions
+
+The `ClusterVersion` object sets the conditions that describe the state of the cluster-version operator (CVO).
+
+Failing:: When `Failing` is set to True, the CVO fails to reconcile the cluster with the desired release image. In all cases, the impact on the cluster is that the dependent nodes in the manifest graph may not be reconciled. The graph may be flattened, in which case there are no dependent nodes.
++
+Most reconciliation errors result in `Failing=True`, even though `ClusterOperatorNotAvaliable` has special handling.
+
+NoDesiredImage:: The CVO has not been given a release image to reconcile. When NoDesiredImage is True, it is a CVO coding error. The `desiredUpdate` field returns you to the current CVO's release image.
+
+ClusterOperatorNotAvailable:: `ClusterOperatorNotAvailable` is set when the CVO fails to retrieve the ClusterOperator from the cluster or when the retrieved ClusterOperator does not satisfy the reconciliation conditions.
```

Comment:
This is being narrowed down since the 4.7 changes, and now some reconciliation failures (like `Degraded=True`) get their own reason (like `ClusterOperatorDegraded`).  As of openshift/cluster-version-operator@0ec39d9b2ab1feee8815d7b6b4bbe2db23daf847, `ClusterOperatorNotAvailable` continues to be used for:

* ClusterOperator manifest has `status.versions` empty.  This should never happen in the wild, and is just there in the code to ensure we fail in CI any pull request that might introduce such a manifest.
* The CVO fails to retrieve the ClusterOperator from the Kube API-server.  Hopefully very rare.
* The ClusterOperator is missing an `Available=True` condition.  Ideally this is very rare, but in reality a number of component operators are a bit twitchy and will go `Available=False` fairly often today.  There are bugs opened for getting these fixed.

Reply:
Closing this since content has drastically changed.